### PR TITLE
make sure we do not mutate the build config in unit test

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/17559

### CODE REVIEW

File: pkg/image/controller/trigger/image_trigger_controller_test.go [hunk 1]
```diff
@@ -936,16 +936,17 @@ func benchmark_1_imageStream(identity, maxTags, sequence int32, round, index int
 // updateBuildConfigImages updates the LastTriggeredImageID field on a build config.
 func updateBuildConfigImages(bc *buildapi.BuildConfig, tagRetriever trigger.TagRetriever) (*buildapi.BuildConfig, error) {
 	var updated *buildapi.BuildConfig
-	for i, t := range bc.Spec.Triggers {
-		p := t.ImageChange
+	bcCopy := bc.DeepCopy()
```
Comment (smarterclayton) [inline pkg/image/controller/trigger/image_trigger_controller_test.go]: The point of this method is to return "new object with no shared memory, or nil".  If bc was mutated, GetInputReference is broken (mutating, and must be fixed) or DeepCopy is broken (mutating, and must be fixed).  I don't see how this change helps

Comment (smarterclayton) [reply pkg/image/controller/trigger/image_trigger_controller_test.go]: The extra deep copies you added in d4859926e1725c6f685e38aeff2dc4acee216111 make no sense now either.

Comment (smarterclayton) [reply pkg/image/controller/trigger/image_trigger_controller_test.go]: I see you didn't add those deep copies - that makes this worse - we did a deep copy before (and weren't seeing these issue?)  And then we update to 1.8 and suddenly we have race conditions?

I'm 90% sure DeepCopy has a race - the other alternative is that the reported race isn't actually the last reader (but I don't see how that could be).

Comment (smarterclayton) [reply pkg/image/controller/trigger/image_trigger_controller_test.go]: We should probably try to test that a random update to a build config doesn't mutate the original.

Comment (mfojtik) [reply pkg/image/controller/trigger/image_trigger_controller_test.go]: cc @sttts i will continue to debug this tomorrow

Comment (mfojtik) [reply pkg/image/controller/trigger/image_trigger_controller_test.go]: also I ran that test 20 times in cycle and was not able to hit the race

**Comment by stevekuznetsov** [comment]
/remove-lifecycle stale
@mfojtik ping

**Comment by mfojtik** [comment]
@stevekuznetsov i think the race was already fixed, we don't need this.

**Pull request closed without merging**